### PR TITLE
Using one alias for one multiple containers [full ci]

### LINKED ISSUE #2592: Not possible to have network alias that is shared among multiple containers
I am trying to get multiple containers to have the same network alias in a single network. This should be supported ([network-scoped alias in Docker documentation](https://docs.docker.com/engine/userguide/networking/work-with-networks/#/network-scoped-alias)). On VIC multiple containers with a single alias can be created, but only one of them is able to start.

**VIC Version:** Jfrog bintray build 5316

**Code (Docker)**

``` bash
docker network create alpine-network
# f2cff761b3925d6537be7726e1fcd55629ced7a8ffabc4ee66be21dda15fc6ca

docker run -itd --name alp-1 --network alpine-network --network-alias alp-app alpine sh
# 961179bbeb9a7c4e3012c0128b8abd3149019ea11aae2a2926326846aaaac37b

docker run -itd --name alp-2 --network alpine-network --network-alias alp-app alpine sh
# 8664a146d1674063901b2470697eae7212ad924786663b6f3d90ead837add654

 docker ps -a
# CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
# 8664a146d167        alpine              "sh"                45 minutes ago      Up 45 minutes                           alp-2
# 961179bbeb9a        alpine              "sh"                45 minutes ago      Up 45 minutes                           alp-1
```

**Code (VIC)**

``` bash
docker --tls network create alpine-network
a7147ac5377dba022519364aa459d1181cef441c6911500deede33ac91ffcc7b

docker --tls run -itd --name alp-1 --network alpine-network --network-alias alp-app alpine sh
# 3118057a742cbe34b41fcd5ef9e686085d5ad8cda66439228717685c9857420e

docker --tls run -itd --name alp-2 --network alpine-network --network-alias alp-app alpine sh
# e1dc071097a84a37bcb11c158c1eab087779c903ea33e75d57f95d5db0b16d46
# C:\Program Files\Docker Toolbox\docker.exe: Error response from daemon: Server error from portlayer: duplicate alias alpine-network:alp-app for container e1dc071097a84a37bcb11c158c1eab087779c903ea33e75d57f95d5db0b16d46.

docker --tls ps -a
# CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
# e1dc071097a8        alpine              "sh"                42 minutes ago      Created                                 alp-2
# 3118057a742c        alpine              "sh"                45 minutes ago      Up 45 minutes                           alp-1

```

**Expected behavior:**
Having both containers started and both of them being aliased as `alp-app`. Also, there should be some built-in behavior that selects which of the containers to respond. The official documentation says

> When multiple containers share the same alias, name resolution to that alias will happen to one of the containers (typically the first container that is aliased). When the container that backs the alias goes down or disconnected from the network, the next container that backs the alias will be resolved.

However, it looks like that the actual case is more complex, possibly a built-in loadbalancer. I have connected a third alpine container to the network and run a few pings to the `alp-app` alias:

``` bash
/ # ping -w 4 alp-app
PING alp-app (172.18.0.2): 56 data bytes
64 bytes from 172.18.0.2: seq=0 ttl=64 time=0.102 ms
64 bytes from 172.18.0.2: seq=1 ttl=64 time=0.167 ms
64 bytes from 172.18.0.2: seq=2 ttl=64 time=0.184 ms
64 bytes from 172.18.0.2: seq=3 ttl=64 time=0.168 ms

--- alp-app ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max = 0.102/0.155/0.184 ms
/ # ping -w 4 alp-app
PING alp-app (172.18.0.2): 56 data bytes
64 bytes from 172.18.0.2: seq=0 ttl=64 time=0.056 ms
64 bytes from 172.18.0.2: seq=1 ttl=64 time=0.160 ms
64 bytes from 172.18.0.2: seq=2 ttl=64 time=0.161 ms
64 bytes from 172.18.0.2: seq=3 ttl=64 time=0.161 ms

--- alp-app ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max = 0.056/0.134/0.161 ms
/ # ping -w 4 alp-app
PING alp-app (172.18.0.3): 56 data bytes
64 bytes from 172.18.0.3: seq=0 ttl=64 time=0.058 ms
64 bytes from 172.18.0.3: seq=1 ttl=64 time=0.107 ms
64 bytes from 172.18.0.3: seq=2 ttl=64 time=0.163 ms
64 bytes from 172.18.0.3: seq=3 ttl=64 time=0.120 ms

--- alp-app ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max = 0.058/0.112/0.163 ms
```

**Actual behavior:**
Only the first container with the `alp-app` network alias is able to start, the second is only created and fails to start with `Server error from portlayer: duplicate alias alpine-network:alp-app for container`

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #2592

### CODE REVIEW

File: lib/dns/dns.go [hunk 1]
```diff
@@ -44,6 +45,10 @@ var (
 	options = ServerOptions{}
 )
 
+func init() {
+	rand.Seed(time.Now().Unix())
```
Comment (vburenin) [inline lib/dns/dns.go]: Are you sure you want to seed a global rand? Have you considered having own instance?
I also would consider using time.Now().UnixNano()

**Comment by caglar10ur** [comment]
LGTM

**Comment by vburenin** [comment]
LGTM

**Comment by hmahmood** [comment]
♻️

**Pull request merged**